packages/skeleton/src/components/skeleton.ts

```diff
@@ -0,0 +1,52 @@
+import path from 'path'
+
+import chalk from 'chalk'
+
+import { RedwoodDiagnostics, RedwoodError, RedwoodWarning } from './diagnostic'
+
+export abstract class RedwoodSkeleton implements RedwoodDiagnostics {
+  warnings: RedwoodWarning[] = []
+  errors: RedwoodError[] = []
+
+  public readonly name: string
+
+  constructor(public readonly filepath: string, name?: string) {
+    this.name = name ?? path.parse(this.filepath).name // default to the file name if not given a specific name
+  }
+
+  hasWarnings(): boolean {
+    return this.warnings.length > 0
+  }
+
+  printWarnings(): void {
+    if (this.warnings.length > 0) {
+      const titleLine = `${chalk.bgYellow('[Warn]')}\t${this.name} ${chalk.dim(
+        this.filepath
+      )}`
+      const warningLines = this.warnings.map((warning) => {
+        return ` (W${warning.code}) ${warning.message}\n`
+      })
+      console.log(titleLine.concat('\n', ...warningLines).trimEnd())
+    }
+  }
+
+  hasErrors(): boolean {
+    return this.errors.length > 0
+  }
+
+  printErrors(): void {
+    if (this.errors.length > 0) {
+      const titleLine = `${chalk.bgRed('[Error]')}\t${this.name} ${chalk.dim(
+        this.filepath
+      )}`
+      const errorLines = this.errors.map((error) => {
+        return ` (E${error.code}) ${error.message}\n`
+      })
+      console.log(titleLine.concat('\n', ...errorLines).trimEnd())
+    }
+  }
+
+  executeAdditionalChecks(): void {
+    // Default implemention should do nothing
+  }
```

Comment:
My thinking was that by providing the default implementation in the superclass then we are able to avoid having to implement it in all subclasses. We can then override it in the subclasses which require it whilst still knowing it's safe to call on any class which inherits from project.